var/spack/repos/builtin/packages/boost/package.py

```diff
@@ -116,7 +116,7 @@ class Boost(Package):
     variant('graph', default=False,
             description="Build the Boost Graph library")
 
-    depends_on('icu', when='+icu_support')
+    depends_on('icu4c', when='+icu_support')
```

Comment:
Any chance you could rename the `icu_support` variant to be consistent with the others in `boost`?  Nothing currently depends on `boost+icu_support`.